frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java

```diff
@@ -587,10 +589,26 @@ private void loadSpdxLicenseInfo(ResourceRequest request, ResourceResponse respo
                     }
                 });
                 jsonGenerator.writeEndArray();
-            } else if (licenseWithText != null) {
+            } else if (mainLicenseNames.size() > 0) {
                 jsonGenerator.writeStringField(LICENSE_PREFIX, LanguageUtil.get(resourceBundle,"main.license.id"));
-                jsonGenerator.writeArrayFieldStart("licenseIds");
-                jsonGenerator.writeString(licenseWithText.getLicenseSpdxId());
+                jsonGenerator.writeArrayFieldStart(LICENSE_IDS);
+                mainLicenseNames.forEach(license -> {
```

Comment:
May be `wrapException(_)` can be used here.

Reply:
done